synapse/storage/databases/main/room.py

```diff
@@ -409,22 +409,32 @@ async def get_rooms_paginate(
             limit: maximum amount of rooms to retrieve
             order_by: the sort order of the returned list
             reverse_order: whether to reverse the room list
-            search_term: a string to filter room names by
+            search_term: a string to filter room names,
+                canonical alias and room ids by
         Returns:
             A list of room dicts and an integer representing the total number of
             rooms that exist given this query
         """
         # Filter room names by a string
         where_statement = ""
+        search_pattern = ""
         if search_term:
-            where_statement = "WHERE LOWER(state.name) LIKE ?"
+            where_statement = """
+                WHERE LOWER(state.name) LIKE ?
+                OR LOWER(state.canonical_alias) LIKE ?
+                OR LOWER(state.room_id) LIKE ?
+            """
 
             # Our postgres db driver converts ? -> %s in SQL strings as that's the
             # placeholder for postgres.
             # HOWEVER, if you put a % into your SQL then everything goes wibbly.
             # To get around this, we're going to surround search_term with %'s
             # before giving it to the database in python instead
-            search_term = "%" + search_term.lower() + "%"
+            search_pattern = [
+                "%" + search_term.lower() + "%",
+                "#%" + search_term.lower() + "%:%",
```

Comment:
this feels a bit redundant. All canonical aliases should start with # and contain a domain part, so why not just:

```suggestion
                "%" + search_term.lower() + "%",
```

Reply:
The reason/idea is to search only in local part and not in server part.
Similar to users list
https://github.com/matrix-org/synapse/blob/66bdca3e317d1fa764cf52547aee7409acc59676/synapse/storage/databases/main/__init__.py#L289-L295

Reply:
I guess I'm not entirely sure we should only search the local part here, but ok.

Reply:
I can change it, if you like.
A good example is: find rooms with "matrix" at matrix.org homeserver. You have no chance to filter and find the "matrix" rooms.

Reply:
fair point!